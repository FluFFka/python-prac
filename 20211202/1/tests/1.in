class AbacabA(int, metaclass=dump):
    def __add__(self, other):
        print("want to add", self, "to", other)
        return self.real+other

class Cab(str, metaclass=dump):
    def __add__(self, other):
        s = ''
        for i in range(other):
            s += self.__str__()
        return s
    def __str__(self):
        return f"<>{super().__str__()}<>"

a = AbacabA(123)
print(a + 1)
b = Cab("chc")
print(b)
print(b + 3)
