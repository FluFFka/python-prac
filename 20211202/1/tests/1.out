__add__: (1,), {}
want to add 123 to 1
124
__str__: (), {}
<>chc<>
__add__: (3,), {}
__str__: (), {}
__str__: (), {}
__str__: (), {}
<>chc<><>chc<><>chc<>
